### DISCUSSION TITLE
How to get text FormData in Adonis 5, while streaming file upload

### DISCUSSION THREAD

**Discussion by D-Marc1** [adonisjs/core#1824] [Help]
In Adonis 4, you used to be able to do the following:

```js
const body = {};

request.multipart.field((name, value) => {
  body[name] = value;
});
```

However, this isn't available in v5. The documentation for v5 states that the body parser can handle multipart/form-data, yet this clearly isn't the case.

```js
// Doesn't work with FormData
const { title } = request.only(['title']);

request.multipart.onFile('file', {}, async (file) => {
  
});

await request.multipart.process();
```

Does anyone know how to use the text FormData with file upload in Adonis 5?

**Comment by Banashek** [comment]
Not sure if this answers your question, but the docs [have an example](https://preview.adonisjs.com/guides/http/file-uploads#using-request-validator) of a form with 2 fields `email` and `avatar`.

It uses the schema to describe the shape of the data (both a `string` and a `file`) and parses that in a single statement.

```ts
public async store ({ request }: HttpContextContract) {
  const userSchema = schema.create({
    email: schema.string(),
    avatar: schema.file({
      size: '2mb',
      extnames: ['jpg', 'png', 'jpeg'],
    }),
  })

  const data = await request.validate({
    schema: userSchema,
  })

  await data.avatar.move(Application.tmpPath('uploads'))
  return 'File uploaded successfully'
}
```

Does this solve for the use-case you're talking about here?

**Reply by D-Marc1** [reply]
I appreciate the attempt, but it doesn't appear that this solves my issue, unfortunately
